spec/unit/command/diff_spec.rb

```diff
@@ -68,6 +69,40 @@
 
     end
 
+    context "when an PolicyfileServiceError is encountered" do
+
+      let(:params) { %w{ --head } }
+
+      context "without a reason" do
+
+        it "prints the exception successfully" do
+          expect(command).to receive(:print_diff).and_raise(ChefDK::PolicyfileServiceError)
+          expect(command.run(params)).to eq(1)
+          expect(ui.output).to include("Error: ChefDK::PolicyfileServiceError")
+        end
+
+      end
+
+      context "with a reason" do
+
+        # Old Ruby versions won't have errors that respond to `reason`,
```

Comment:
And `#reason` is just a thing I made up for our custom nested exceptions to avoid conflicting with the ruby nested exceptions.